### PR TITLE
Fixes an exception, if a mail has no dkim signature (there's no sdid)

### PR DESCRIPTION
**Pull request by dodmi**

### CODE REVIEW

File: modules/dkimPolicy.jsm [hunk 1]
```diff
@@ -525,6 +525,7 @@ var Policy = {
 			}
 
 			// Check if enabled for SDID.
+			if (!sdid) { return undefined; }
```
Comment (lieser) [inline modules/dkimPolicy.jsm:528]: Not that on master this check is done in `addFavicons()` in `AuthVerifier.jsm`.

I'm fine with leaving the check here, but then you should adapt the type of the `sdid` parameter to `String|undefined`.

**Review by lieser — COMMENTED** [review]
Note that all this type hints in the code are meant to help catch such errors by using the Type Checking feature of the Type Script compiler https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html.

I don't know in how good a state this is for the 2.x. branch. Depending on how long you still plan to maintain the 2.x. branch, and how much you are interested in using such code analysis to help developing, it may be worth setting up a local environment that supports this. I myself am using VS Code.

Let me know if you are interested, I could help you out a little to get this in a somehow working state.

A little off topic, but wondering that keeps you on such an old Thunderbird version?

File: modules/dkimPolicy.jsm (hunk 1 above)
Comment (dodmi) [reply modules/dkimPolicy.jsm]: Thank you for your guidance, again. I moved the check to AuthVerifier.jsm, in the end.
I guess, adding/updating type checks to the 2.x branch would be a lot of work, as I didn't really care about them. So thank you for your kind offer. But I'm not sure, I want to do this at the moment.

Off topic, but no problem:
- I like my configuration of Thunderbird with all its addons and the theme, as it is. At the moment I'm willing to put some effort in improving it. I've fixed numerous issues, which came to my attention. But mostly for addons, where the developer lost his/her interest in keeping up with the changes in Thunderbird. 
- I learn a lot, while I'm doing this (which would be a point on the pro site of adding type checking to the 2.x branch)
- I know, there are more and more security issues on the other hand. This is the only reason I can think of, at the moment, which could bring me to switch to a current mail client.
- But changing the mail client wouldn't be easy; probably I wouldn't find a client with all the features, I'm used to. There aren't many mail clients on windows, so the options are pretty limited. The next redesign for the current Thunderbird is already announced - but probably the developers will announce another redesign before I can invest some time to try to get it in a usable state (usable in my eyes)

**Review by lieser — APPROVED** [review]

**Comment by dodmi** [comment]
> I don't know in how good a state this is for the 2.x. branch. Depending on how long you still plan to maintain the 2.x. branch, and how much you are interested in using such code analysis to help developing, it may be worth setting up a local environment that supports this. I myself am using VS Code.

Well, since I pretty ignored type hints until now, I guess, it would be a lot of work. I don't know, if it's worth the time.

> Let me know if you are interested, I could help you out a little to get this in a somehow working state.

Thank you for your kind offer, but I guess, I won't do it for now.

> A little off topic, but wondering that keeps you on such an old Thunderbird version?

Old doesn't mean bad and new isn't good in all cases.
I'm happy with my configuration, which got better and better for years. 
I couldn't update to a later Thunderbird version without a loss of functionality. The next redesign for Thunderbird is already anounced...
On the other hand, there are more and more security issues, which is the only thing making me switch.
But then, I need to find a replacement for my old Thunderbird, which probably wouldn't be the latest Thunderbird.

**Comment by lieser** [comment]
Thanks for the explanation.

Note that your first comment did not disappear, it is just in the resolved conversation that is hidden by default.

**Comment by dodmi** [comment]
Ah, ok - it was pretty late the day before yesterday :-)

**Pull request merged**